From 8dbe877ae9f7f89709b7a3868af1ca1cdc2e0661 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 15 Nov 2021 14:20:23 +0200
Subject: [PATCH] [SW-60588] gaudi2: add pcie lbw/hbw err indication

As a part of pcie RR razwi occurrence validation part, the scratchpad
is updated when the indication is set. But as the security
configuration locks the indication register, it cannot be cleared by
validation, from outside the firmware. Hence this patch clears the
indication inside firmware.

This patch will be reverted post Gaudi2-PO validation activity.

Change-Id: I038b420084d85747b1eee278dd5383654ed0770a
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index e61d9dd..7f69dca 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1028,9 +1028,12 @@
  */
 void check_razwi_intr(void)
 {
-	u32 reg = 0;
+	u32 reg = regmap_readl(mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
+	u32 bitmask = reg & PCIE_AXI_ERR_INTR_BIT;
 
-	reg = regmap_readl(mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
-	if (reg & PCIE_AXI_ERR_INTR_BIT)
+	if (bitmask) {
 		regmap_writel(RAZWI_INTR_STATUS_PATTERN, mmGIC_RAZWI_STATUS_REG);
+		/* Reset the indication */
+		regmap_writel(reg & ~bitmask, mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
+	}
 }
